File: Java/RecyclerView14RecyclerViewDragAndSwap/app/src/main/java/dev/ogabek/recyclerview14_recyclerviewdragandswap/activity/MainActivity.java
package dev.ogabek.recyclerview14_recyclerviewdragandswap.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.os.Bundle;
import android.view.animation.AnimationUtils;
import android.view.animation.LayoutAnimationController;

import java.util.ArrayList;
import java.util.List;

import dev.ogabek.recyclerview14_recyclerviewdragandswap.R;
import dev.ogabek.recyclerview14_recyclerviewdragandswap.adapter.MainRecyclerViewAdapter;
import dev.ogabek.recyclerview14_recyclerviewdragandswap.helper.SimpleItemTouchHelperCallback;
import dev.ogabek.recyclerview14_recyclerviewdragandswap.model.Member;

public class MainActivity extends AppCompatActivity {

    Context context;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        refreshAdapter(prepareMemberList());

    }

    private List<Member> prepareMemberList() {
        List<Member> members = new ArrayList<>();
        for (int i = 0; i < 50; i++) {
            members.add(new Member(String.valueOf(i)));
        }
        return members;
    }

    private void initViews() {
        context = this;
        recyclerView = findViewById(R.id.rv_mainRecyclerView);
        recyclerView.setLayoutManager(new GridLayoutManager(context, 1));
    }

    private void refreshAdapter(List<Member> members) {
        MainRecyclerViewAdapter adapter = new MainRecyclerViewAdapter(context, members);
        recyclerView.setAdapter(adapter);
        ItemTouchHelper.Callback callback = new SimpleItemTouchHelperCallback(adapter);
        ItemTouchHelper touchHelper = new ItemTouchHelper(callback);
        touchHelper.attachToRecyclerView(recyclerView);
    }
}

File: Kotlin/RecyclerView14RecyclerViewDragAndSwipe/app/src/main/java/dev/ogabek/recyclerview_14_recyclerviewdragandswipe/activity/MainActivity.kt
package dev.ogabek.recyclerview_14_recyclerviewdragandswipe.activity

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.recyclerview_14_recyclerviewdragandswipe.R
import dev.ogabek.recyclerview_14_recyclerviewdragandswipe.adapter.MainAdapter
import dev.ogabek.recyclerview_14_recyclerviewdragandswipe.helper.SimpleItemTouchHelperCallBack
import dev.ogabek.recyclerview_14_recyclerviewdragandswipe.model.Member

class MainActivity : AppCompatActivity() {
    private lateinit var context: Context
    private lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

        refreshAdapter(prepareMemberList())

    }

    private fun initViews() {
        context = this
        recyclerView = findViewById(R.id.rv_main_recycler_view)
        recyclerView.layoutManager = GridLayoutManager(context, 1)
    }

    private fun refreshAdapter(members: List<Member>) {
        val adapter = MainAdapter(context, members as ArrayList<Member>)
        recyclerView.adapter = adapter
        val callback = SimpleItemTouchHelperCallBack(adapter)
        val touchHelper = ItemTouchHelper(callback)
        touchHelper.attachToRecyclerView(recyclerView)
    }

    private fun prepareMemberList(): List<Member> {
        val members = ArrayList<Member>()
        for (i in 0..50) members.add(Member("$i"))
        return members
    }

}